scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -87,6 +87,33 @@ impl ReplicatorConsumer {
         self.update_or_insert(data, true).await;
     }
 
+    // Recreates INSERT/DELETE statement on non-frozen map.
+    async fn update_map<'a>(
+        &self,
+        column_name: &str,
+        data: &'a CDCRow<'_>,
+        values_for_update: &mut Vec<&'a CqlValue>,
+        values_for_delete: &[&CqlValue],
+        timestamp: i64,
+    ) {
+        let deleted_elements: &'a [CqlValue] = data.get_deleted_elements(column_name);
+
+        if (data.get_value(column_name).is_none() && deleted_elements.is_empty())
+            || data.is_value_deleted(column_name)
+        {
+            // INSERT/DELETE/OVERWRITE
+            self.update_atomic_or_frozen(
```

Comment:
That's very confusing, since both the name and comment on `update_atomic_or_frozen` suggest that it can only be used for atomic/frozen types, and here you're using it for a non-frozen collection.

The function's contract should be improved (generalized).

Reply:
Also, `update_atomic_or_frozen` will replicate collection-wide tombstones using `DELETE` statements, which will use a wrong timestamp. You need to either adjust the timestamp, or change `update_atomic_or_frozen` to replicate deletes using `UPDATE` (`SET v = null`).

Read this:
https://docs.scylladb.com/using-scylla/cdc/cdc-advanced-types/#collection-wide-tombstones-and-timestamps

Reply:
I have implemented second option and I think it should be correct now.